### ISSUE TITLE
Inconsistent partial lookup between PartialRenderer and Digestor

### ISSUE THREAD

**Issue by zhengpd** [rails/rails#30131]
The partial renderer will use `@lookup_context.prefixes` to find template:

```ruby
# https://github.com/rails/rails/blob/master/actionview/lib/action_view/renderer/partial_renderer.rb#L421
def find_template(path, locals)
  prefixes = path.include?(?/) ? [] : @lookup_context.prefixes
  @lookup_context.find_template(path, prefixes, true, locals, @details)
end
```

 but in Digestor the template is found with empty prefixes argument: 

```ruby
# https://github.com/rails/rails/blob/master/actionview/lib/action_view/digestor.rb#L52
template = finder.disable_cache { finder.find_all(logical_name, [], partial, [], options).first }
```

This seems inconsistent. It can cause `Couldn't find template for digesting` error when caching is enabled and `lookup_context.prefixes` is modified on controller level like how camaleon-cms does in its [FrontendController](https://github.com/owen2345/camaleon-cms/blob/master/app/controllers/camaleon_cms/frontend_controller.rb#L222).

**Comment by rafaelfranca** [comment]
Is this causing any issue?

**Comment by zhengpd** [comment]
@rafaelfranca No. I came across this when working my project and thought there might be potential improvement. Close the issue if you think it's meaningless.

**Comment by rafaelfranca** [comment]
I think it makes sense to be consistent. Do you want to work in a PR?

**Comment by st0012** [comment]
I can open an PR for this

**Comment by zhengpd** [comment]
@st0012 That's great 👍 . Sorry I don't have time for this.

**Comment by st0012** [comment]
Those code are still the same. I'll try to finish my PR.

**Comment by st0012** [comment]
@rafaelfranca I think I found the reason we only pass an empty array in the Digester. It's because the `logical_name` already contains the template's prefixes.
And I don't think it's a good idea changing this behavior just for consistency, because that requires large change on `Digester.digest`'s implementation (I guess). I suggest that we can put a comment above the code to explain why the prefixes are empty.
